### PR TITLE
Update creating-a-github-pages-site-with-jekyll.md

### PR DESCRIPTION
**Pull request by gojimmypi**
add notes for `domain`, `url`, and `baseurl` settings when repo used as subfolder

### Why:

adds important, missing information

### What's being changed:

minor documentation & config example

### CODE REVIEW

**Comment by janiceilene** [comment]
@gojimmypi Thanks so much for opening a PR! I'll get this triaged for review ⚡

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md [hunk 1]
```diff
@@ -83,6 +83,14 @@ Before you can use Jekyll to create a {% data variables.product.prodname_pages %
 
    Replace _GITHUB-PAGES-VERSION_ with the latest supported version of the `github-pages` gem. You can find this version here: "[Dependency versions](https://pages.github.com/versions/)."
 
+   Make any necessary edits to the `_config.yml` file (needed for relative paths when the [repo is hosted in subdirectory](https://docs.github.com/en/github/getting-started-with-github/using-git/splitting-a-subfolder-out-into-a-new-repository)):
```
Comment (mchammer01) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:86]:
```suggestion
   Make any necessary edits to the `_config.yml` file. This is required for relative paths when the repository is hosted in a subdirectory.  For more information, see "[Splitting a subfolder out into a new repository](/github/getting-started-with-github/using-git/splitting-a-subfolder-out-into-a-new-repository)."
```

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md [hunk 2]
```diff
@@ -83,6 +83,14 @@ Before you can use Jekyll to create a {% data variables.product.prodname_pages %
...
+   
+```yml
+domain: my-site.github.io       # if you want to force HTTPS, put down the domain without the http: e.g. example.com
```
Comment (mchammer01) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:89]:
```suggestion
domain: my-site.github.io       # if you want to force HTTPS, specify the domain without the http at the start, e.g. example.com
```

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md [hunk 3]
```diff
@@ -83,6 +83,14 @@ Before you can use Jekyll to create a {% data variables.product.prodname_pages %
...
+url: https://my-site.github.io  # the base hostname & protocol for your site, e.g. http://example.com
```
Comment (mchammer01) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:90]:
```suggestion
url: https://my-site.github.io  # the base hostname and protocol for your site, e.g. http://example.com
```

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md [hunk 4]
```diff
@@ -83,6 +83,14 @@ Before you can use Jekyll to create a {% data variables.product.prodname_pages %
...
+baseurl: /REPOSITORY-NAME/      # place folder name if site is served in subfolder
```
Comment (mchammer01) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:91]:
```suggestion
baseurl: /REPOSITORY-NAME/      # place folder name if the site is served in a subfolder
```

**Review by mchammer01 — COMMENTED** [review]
@gojimmypi 👋🏻 - thank you so much for helping us improve our docs ✨ 
I've made a few suggestions from an editorial perspective, and I'm going to apply these suggestions.
I've asked someone to double-check the technical accuracy of the updates as I am not familiar with GitHub Pages.

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md [hunk 5]
```diff
@@ -83,6 +83,14 @@ Before you can use Jekyll to create a {% data variables.product.prodname_pages %
...
    Replace _GITHUB-PAGES-VERSION_ with the latest supported version of the `github-pages` gem. You can find this version here: "[Dependency versions](https://pages.github.com/versions/)."
 
+   Make any necessary edits to the `_config.yml` file. This is required for relative paths when the repository is hosted in a subdirectory.  For more information, see "[Splitting a subfolder out into a new repository](/github/getting-started-with-github/using-git/splitting-a-subfolder-out-into-a-new-repository)."
+   
+```yml
+domain: my-site.github.io       # if you want to force HTTPS, specify the domain without the http at the start, e.g. example.com
+url: https://my-site.github.io  # the base hostname and protocol for your site, e.g. http://example.com
+baseurl: /REPOSITORY-NAME/      # place folder name if the site is served in a subfolder
+```
+
```
Comment (mchammer01) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:93]:
```suggestion
```

**Comment by mchammer01** [comment]
👋🏻 - there is an issue with the ordered list that I'll need to investigate.

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md [hunk 6]
```diff
@@ -83,6 +83,13 @@ Before you can use Jekyll to create a {% data variables.product.prodname_pages %
 
    Replace _GITHUB-PAGES-VERSION_ with the latest supported version of the `github-pages` gem. You can find this version here: "[Dependency versions](https://pages.github.com/versions/)."
 
+   Make any necessary edits to the `_config.yml` file. This is required for relative paths when the repository is hosted in a subdirectory.  For more information, see "[Splitting a subfolder out into a new repository](/github/getting-started-with-github/using-git/splitting-a-subfolder-out-into-a-new-repository)."
+   
+```yml
```
Comment (mchammer01) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:88]:
```suggestion
   ```yml
```

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md [hunk 7]
```diff
@@ -83,6 +83,13 @@ Before you can use Jekyll to create a {% data variables.product.prodname_pages %
...
+domain: my-site.github.io       # if you want to force HTTPS, specify the domain without the http at the start, e.g. example.com
```
Comment (mchammer01) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:89]:
```suggestion
   domain: my-site.github.io       # if you want to force HTTPS, specify the domain without the http at the start, e.g. example.com
```

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md [hunk 8]
```diff
@@ -83,6 +83,13 @@ Before you can use Jekyll to create a {% data variables.product.prodname_pages %
...
+url: https://my-site.github.io  # the base hostname and protocol for your site, e.g. http://example.com
```
Comment (mchammer01) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:90]:
```suggestion
   url: https://my-site.github.io  # the base hostname and protocol for your site, e.g. http://example.com
```

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md [hunk 9]
```diff
@@ -83,6 +83,13 @@ Before you can use Jekyll to create a {% data variables.product.prodname_pages %
...
+baseurl: /REPOSITORY-NAME/      # place folder name if the site is served in a subfolder
```
Comment (mchammer01) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:91]:
```suggestion
   baseurl: /REPOSITORY-NAME/      # place folder name if the site is served in a subfolder
```

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md [hunk 10]
```diff
@@ -83,6 +83,13 @@ Before you can use Jekyll to create a {% data variables.product.prodname_pages %
...
+```
```
Comment (mchammer01) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:92]:
```suggestion
   ```
```

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md [hunk 11]
```diff
@@ -83,6 +83,13 @@ Before you can use Jekyll to create a {% data variables.product.prodname_pages %
...
    The correct version Jekyll will be installed as a dependency of the `github-pages` gem.
 10. Save and close the Gemfile.
```
Comment (mchammer01) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:94]:
```suggestion
1. Save and close the Gemfile.
```

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md [hunk 12]
```diff
@@ -83,6 +83,13 @@ Before you can use Jekyll to create a {% data variables.product.prodname_pages %
 
    Replace _GITHUB-PAGES-VERSION_ with the latest supported version of the `github-pages` gem. You can find this version here: "[Dependency versions](https://pages.github.com/versions/)."
 
+   Make any necessary edits to the `_config.yml` file. This is required for relative paths when the repository is hosted in a subdirectory.  For more information, see "[Splitting a subfolder out into a new repository](/github/getting-started-with-github/using-git/splitting-a-subfolder-out-into-a-new-repository)."
+
```
Comment (mchammer01) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:87]:
```suggestion
```

**Comment by mchammer01** [comment]
I've fixed the issues with the ordered list restarting at one, and several list items being in the same paragraph 🙂 
Now just waiting on someone to check that the updates are accurate from a technical point of view.

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md [hunk 13]
```diff
@@ -83,8 +83,13 @@ Before you can use Jekyll to create a {% data variables.product.prodname_pages %
...
    Replace _GITHUB-PAGES-VERSION_ with the latest supported version of the `github-pages` gem. You can find this version here: "[Dependency versions](https://pages.github.com/versions/)."
 
+   Make any necessary edits to the `_config.yml` file. This is required for relative paths when the repository is hosted in a subdirectory.  For more information, see "[Splitting a subfolder out into a new repository](/github/getting-started-with-github/using-git/splitting-a-subfolder-out-into-a-new-repository)."
+   ```yml
+   domain: my-site.github.io       # if you want to force HTTPS, specify the domain without the http at the start, e.g. example.com
+   url: https://my-site.github.io  # the base hostname and protocol for your site, e.g. http://example.com
+   baseurl: /REPOSITORY-NAME/      # place folder name if the site is served in a subfolder
    The correct version Jekyll will be installed as a dependency of the `github-pages` gem.
-10. Save and close the Gemfile.
+1. Save and close the Gemfile.
```
Comment (YiMysty) [inline content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:92]:
```suggestion
10. Save and close the Gemfile.
```
should this be 10? :D

**Review by jriggins — COMMENTED** [review]
Some ordering suggestions based on what I _think_ is being added.

File: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md (hunk 13 above)
Comment (mchammer01) [reply content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md:92]: The numbering doesn't matter in the source file (we could use `1.` for all entries), this is processed at building time. The reason I put 1. here was to try and solve some issues with the list.

**Review by mchammer01 — APPROVED** [review]
@gojimmypi - LGTM ✨

**Comment by mchammer01** [comment]
Thank you for your reviews @jriggins and @YiMysty - very much appreciated 💖 (I wouldn't worry about the item list numbering for now, that's something we'll need to sort out in the whole of our docs).

**Pull request merged**